GitCommands/Git/GitModule.cs

```diff
@@ -1367,6 +1380,75 @@ public string ResetFiles(IReadOnlyList<string> files)
                 .BuildBatchArgumentsForFiles(files));
         }
 
+        /// <summary>
+        /// Reset changes to the selected files.
+        /// </summary>
+        /// <param name="selectedItems">Items to reset, with Staged status set.</param>
+        /// <param name="resetAndDelete">Delete new (and renamed) files.</param>
+        /// <param name="fullPathResolver">The </param>
+        /// <param name="filesInUse">Out put listing files in use, that cannot be deleted.</param>
+        /// <param name="output">Error messages from the reset.</param>
+        /// <param name="progressAction">Action when unstaging files (to update a progress bar).</param>
+        /// <returns><see langword="true"/> if successfully executed</returns>
+        public bool ResetChanges(IEnumerable<GitItemStatus> selectedItems, bool resetAndDelete, IFullPathResolver fullPathResolver, out List<string> filesInUse, out StringBuilder output, Action<BatchProgressEventArgs>? progressAction = null)
+        {
+            // If Staged was selected, unstage file first
+            IEnumerable<GitItemStatus> stagedFiles = selectedItems.Where(item => item.Staged == StagedStatus.Index).ToList();
```

Comment:
I think `ToList` is superfluous here, `BatchUnstageFiles` takes an enumerable and then materialises it itself.

Reply:
removed (may be needed in #10980)